libs/pageserver_api/src/key.rs

```diff
@@ -564,35 +574,36 @@ pub const TWOPHASEDIR_KEY: Key = Key {
 };
 
 #[inline(always)]
-pub fn twophase_file_key(xid: TransactionId) -> Key {
+pub fn twophase_file_key(xid: u64) -> Key {
```

Comment:
I actually started with that, but found that FullTransaction is a bit awkward to use, because it maps to the FullTransactionId struct in Postgres:
```
#[repr(C)]
#[derive(Debug, Copy, Clone, Default, Serialize, Deserialize)]
pub struct FullTransactionId {
    pub value: uint64,
}
```
Postgres does that to get a little type-safety in C, but it's not great in Rust code.

One option would be to replace that in Rust with something like this instead of letting bindgen generate it:
```
pub type FullTransactionId = uint64;
```
I think that would be compatible, and more convenient to use.

Aside from the convenience, another reason is that on v16 and below, we're still dealing with TransactionIds. So in this function, for example, 'xid' really a TransactionId or a FullTransactionId depending on the Postgres version, but I wanted to avoid the code churn of having two functions. We could still just leave the high bits 0 in the FullTransactionId too, but at that point, is FullTransactionId really correct either?

in the end, I decided to cop out and use u64. Do you think FullTransactionId would still be better? Or any other ideas?